internal/engine/ingester/diff/diff.go

```diff
@@ -203,6 +208,124 @@ func (di *Diff) ingestFileForDepDiff(
+				Dep: &pbinternal.Dependency{
+					Ecosystem: inventoryToEcosystem(inventory),
+					Name:      inventory.Name,
+					Version:   inventory.Version,
+				},
+				File: &pbinternal.PrDependencies_ContextualDependency_FilePatch{
+					Name:     filename,
+					PatchUrl: "", // TODO: do we need this?
+				},
+			})
+		}
+	}
+
+	return &engif.Result{Object: &deps, Checkpoint: checkpoints.NewCheckpointV1Now()}, nil
+}
+
+func inventorySorter(a *extractor.Inventory, b *extractor.Inventory) int {
+	return cmp.Or(cmp.Compare(a.Name, b.Name), cmp.Compare(a.Version, b.Version))
+}
+
+func (di *Diff) scalibrInventory(ctx context.Context, repoURL string, ref string) ([]*extractor.Inventory, error) {
+	clone, err := di.cli.Clone(ctx, repoURL, ref)
+	if err != nil {
+		return nil, err
+	}
+
+	tree, err := clone.Worktree()
+	if err != nil {
+		return nil, err
+	}
+	return scanFs(ctx, tree.Filesystem, map[string]string{})
+}
+
+func scanFs(ctx context.Context, memFS billy.Filesystem, _ map[string]string) ([]*extractor.Inventory, error) {
+	// have to down-cast here, because scalibr needs multiple io/fs types
+	wrapped, ok := iofs.New(memFS).(scalibr_fs.FS)
+	if !ok {
+		return nil, fmt.Errorf("error converting filesystem to ReadDirFS")
+	}
+
+	desiredCaps := scalibr_plugin.Capabilities{
+		OS:            scalibr_plugin.OSAny,
+		Network:       true,
+		DirectFS:      false,
+		RunningSystem: false,
+	}
+
+	scalibrFs := scalibr_fs.ScanRoot{FS: wrapped}
+	scanConfig := scalibr.ScanConfig{
+		ScanRoots: []*scalibr_fs.ScanRoot{&scalibrFs},
+		// All includes Ruby, Dotnet which we're not ready to test yet, so use the more limited Default set.
+		FilesystemExtractors: list.FilterByCapabilities(list.Default, &desiredCaps),
+		Capabilities: &desiredCaps,
+		Stats: scalibr_stats.NoopCollector{},
+	}
+
+	scanner := scalibr.New()
+	scanResults := scanner.Scan(ctx, &scanConfig)
+
+	if scanResults == nil || scanResults.Status == nil {
+		return nil, fmt.Errorf("error scanning files: no results")
+	}
+	if scanResults.Status.Status != scalibr_plugin.ScanStatusSucceeded {
+		return nil, fmt.Errorf("error scanning files: %s", scanResults.Status)
+	}
+
+	return scanResults.Inventories, nil
+}
+
+func inventoryToEcosystem(inventory *extractor.Inventory) pbinternal.DepEcosystem {
+	if inventory == nil {
+		zerolog.Ctx(context.Background()).Warn().Msg("nil ecosystem scanning diffs")
+		return pbinternal.DepEcosystem_DEP_ECOSYSTEM_UNSPECIFIED
+	}
+	// This should be inventory.GetEcosystem()... but there isn't a convenience wrapper yet
+	ecosystem, err := inventory.Extractor.Ecosystem(inventory)
+	if err != nil {
+		zerolog.Ctx(context.Background()).Warn().Err(err).Msg("error getting ecosystem from inventory")
+		return pbinternal.DepEcosystem_DEP_ECOSYSTEM_UNSPECIFIED
+	}
+
+	// Sometimes Scalibr uses the string "PyPI" instead of "pypi" when reporting the ecosystem.
+	switch strings.ToLower(ecosystem) {
```

Comment:
If you're interested in the PURL-defined package type you should use converter.ToPURL(inventory).Type or inventory.Extractor.ToPURL(inventory).Type (https://github.com/google/osv-scalibr/blob/main/converter/converter.go#L45)

Ecosystem is specific to the OSV schema and sometimes use different strings than PURL package types.

Reply:
Hm, we should probably provide an Inventory.PURL() member function similar to Inventory.Ecosystem() so people don't have to import converter.go.

Reply:
Right, at the moment we don't really have any cases where we'd need to return an error. I'll discuss internally if we ever would want to do that or if we can just remove it.

Reply:
> I'll discuss internally if we ever would want to do that or if we can just remove it.

Update: We ended up removing it: https://github.com/google/osv-scalibr/commit/9f86538b542446508e8fab14eff6eaab11c6e913

So if you pin SCALIBR to after this commit you should be able to get rid of the error handling in your code too.